addons/pos_self_order/models/res_config_settings.py

```diff
@@ -74,6 +78,68 @@ def custom_link_action(self):
             "domain": ['|', ['pos_config_ids', 'in', self.pos_config_id.id], ["pos_config_ids", "=", False]],
         }
 
+    def generate_qr_codes_zip(self):
+        if self.pos_self_ordering_mode == 'mobile' and self.pos_module_pos_restaurant:
+            table_ids = self.pos_config_id.floor_ids.table_ids
+
+            if not table_ids:
+                raise ValidationError(_("In Self-Order mode, you must have at least one table to generate QR codes"))
+
+            urls = [{
+                'url': self.pos_config_id._get_self_order_url(table.id),
+                'name': f"{table.name} - {table.floor_id.name}"
+            } for table in table_ids]
+        else:
+            urls = [{
+                'url': self.pos_config_id._get_self_order_url(),
+                'name': "generic"
+            }]
+
+        qr_images = []
+        for url in urls:
+            qr = qrcode.QRCode(
+                version=1,
+                error_correction=qrcode.constants.ERROR_CORRECT_L,
+                box_size=10,
+                border=4,
+            )
+            qr.add_data(url['url'])
+            qr.make(fit=True)
+            qr_images.append({
+                'image':qr.make_image(fill_color="black", back_color="white"),
+                'name': url['name'],
+            })
+
+        # Create a zip with all images in qr_images
+        zip_buffer = BytesIO()
+        with zipfile.ZipFile(zip_buffer, "w", zipfile.ZIP_DEFLATED) as zip_file:
+            for index, qr_image in enumerate(qr_images):
+                image_buffer = BytesIO()
+                qr_image['image'].save(image_buffer, format="PNG")
+                image_data = image_buffer.getvalue()
+                zip_file.writestr(f"{qr_image['name']}_{index + 1}.png", image_data)
+        zip_buffer.seek(0)
+
+        # Delete previous attachments
+        self.env["ir.attachment"].search([
+            ("name", "=", "self_order_qr_code.zip"),
+        ]).unlink()
```

Comment:
The idea is never to keep more than one zip file in the database, as these may be "outdated". Indeed, table URLs can change if the user resets the "identifiers" and instead of checking whether these "identifiers" have changed since the last zip generation, we generate all QR codes on the fly each time.

Note that this request will be quite rare, as the end customer will need to download this file very rarely.